deploy/templates/consumption-azuredeploy.json

```diff
@@ -351,74 +355,64 @@
+      "name": "[concat(variables('eventhub_namespace_name'), '/devicedata/$Default')]",
       "location": "[parameters('ResourceLocation')]",
       "dependsOn": [
         "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'devicedata')]",
-        "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'normalizeddata')]",
         "[resourceId('Microsoft.EventHub/namespaces', variables('eventhub_namespace_name'))]",
-        "[resourceId('Microsoft.EventHub/namespaces/AuthorizationRules', variables('eventhub_namespace_name'), 'RootManageSharedAccessKey')]"
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'devicedata', 'writer')]",
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'normalizeddata', 'reader')]"
       ],
       "properties": {
-        "rights": [
-          "Listen"
-        ]
       }
     },
     {
-      "type": "Microsoft.EventHub/namespaces/eventhubs/authorizationRules",
+      "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
       "apiVersion": "2017-04-01",
-      "name": "[concat(variables('eventhub_namespace_name'), '/normalizeddata/writer')]",
+      "name": "[concat(variables('eventhub_namespace_name'), '/normalizeddata/$Default')]",
       "location": "[parameters('ResourceLocation')]",
       "dependsOn": [
-        "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'devicedata')]",
         "[resourceId('Microsoft.EventHub/namespaces/eventhubs', variables('eventhub_namespace_name'), 'normalizeddata')]",
         "[resourceId('Microsoft.EventHub/namespaces', variables('eventhub_namespace_name'))]",
-        "[resourceId('Microsoft.EventHub/namespaces/AuthorizationRules', variables('eventhub_namespace_name'), 'RootManageSharedAccessKey')]"
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'devicedata', 'writer')]",
+        "[resourceId('Microsoft.EventHub/namespaces/eventhubs/authorizationRules', variables('eventhub_namespace_name'), 'normalizeddata', 'reader')]"
       ],
       "properties": {
-        "rights": [
-          "Send"
-        ]
       }
     },
     {
-      "type": "Microsoft.EventHub/namespaces/eventhubs/consumergroups",
-      "apiVersion": "2017-04-01",
-      "name": "[concat(variables('eventhub_namespace_name'), '/devicedata/$Default')]",
-      "location": "[parameters('ResourceLocation')]",
+      "type": "Microsoft.EventHub/namespaces/eventhubs/providers/roleAssignments",
+      "apiVersion": "2018-01-01-preview",
+      "name": "[concat(variables('devicedata_eventhub_name'), '/Microsoft.Authorization/', guid(uniqueString(variables('devicedata_eventhub_name'))))]",
```

Comment:
Confused why we need a guid wrapped over unique string for the authorization rule?  Can't it just be a known name?

Reply:
This is just why the documentation instructed - I didn’t try to use anything else once I got everything working.

https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-template#resource